static/js/lib/api.js

```diff
@@ -6,6 +6,92 @@ import R from "ramda"
 
 import { S, parseJSON, filterE } from "./sanctuary"
 
+export const booksRequest = (force = false) => ({
+  url: "/api/library/books/",
+  update: {
+    books: (prev, next) => {
+      return next
+    },
+  },
+  transform: (responseJson) => {
+    // The response is a list of serialized books. This turns it into an object mapping
+    // the book id to the full serialized object: {[book.id]: [book]}
+    return {
+      books: R.fromPairs(
+        responseJson.map((book) => (
+          [book.id, book]
+        ))
+      )
+    }
+  },
+  force
+})
+
+export const deleteBookRequest = (bookId) => ({
+  url: `/api/library/books/${bookId}/`,
+  update:  {
+    books: (prev, next) => {
+      return R.omit(
+        [next.toDelete],
+        prev
+      )
+    },
+  },
+  transform: () => {
+    return {
+      books: {
+        toDelete: String(bookId)
+      }
+    }
+  },
+  options: {
+    method: "DELETE"
+  }
+})
+
+export const bookRequest = (bookId, force = false) => ({
+  url: `/api/library/books/${bookId}/`,
+  update: {
+    book: (prev, next) => {
+      return next
+    },
+  },
+  transform: (responseJson) => {
+    return { book: responseJson }
+  },
+  force
+})
```

Comment:
Obviously there's a lot of boilerplate than we'd want, but a lot of this could be easily abstracted in the same way the `redux-hammock` does it. If we know the name of the resource and the URL for that resource in the API, we can generate most of this.

Reply:
yeah it would be nice to define something to generate a 'default' object like this, which we could then override if we needed to do anything more fancy with a particular endpoint.